src/github.rs

```diff
@@ -110,12 +122,13 @@ impl GithubClient {
                 .client
                 .execute(
                     self.client
-                        .get("https://api.github.com/rate_limit")
+                        .get(&format!("{}/rate_limit", self.api_url))
                         .configure(self)
                         .build()
                         .unwrap(),
                 )
                 .await?;
+            rate_resp.error_for_status_ref()?;
```

Comment:
Note: this call to `error_for_status_ref()` is a small unrelated fix to fail early if the `/rate_limit` URL fails. Otherwise this was ignoring the status code, which can cause a confusing error.